name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java

```diff
@@ -172,206 +174,134 @@ private void addBuySellTransaction()
-
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundDifferentCurrency)
 
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency").optional()
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
-                .assign((t, v) -> {
-                    if (t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundSameCurrency)
 
                 // Devisenkurs   : 1,192200(x)             Provision     :
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency", "exchangeRate").optional()
                 .match("^Devisenkurs ([:\\s]+)?(?<exchangeRate>[.,\\d]+).*$")
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
+                .assign(this::extractTaxRefundDifferentCurrency)
+
+                .wrap(BuySellEntryItem::new);
+
+        addTaxesSectionsTransaction(pdfTransaction, type);
+        addFeesSectionsTransaction(pdfTransaction, type);
+        addTaxReturnBlock(type);
+
+        pdfTransaction
+                // Storno Wertpapierabrechnung Kauf Fonds/Zertifikate
+                // Stornierung Wertpapierabrechnung Kauf Fonds
+                .section().optional()
+                .match("^(Storno|Stornierung) Wertpapierabrechnung.*$")
                 .assign((t, v) -> {
-                    if (!t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        Money taxRefundFX = Money.of(asCurrencyCode(v.get("currency")), asAmount(v.get("taxRefund")));
+                    invertFeesForRefund((BuySellEntry) t);
+                });
+    }
 
-                        BigDecimal exchangeRate = asExchangeRate(v.get("exchangeRate"));
-                        exchangeRate = BigDecimal.ONE.divide(exchangeRate, 10, RoundingMode.HALF_DOWN);
+    private void invertFeesForRefund(BuySellEntry t) {
+        if (t.getPortfolioTransaction().getType() == PortfolioTransaction.Type.SELL) {
+            t.setType(PortfolioTransaction.Type.BUY);
+        } else if(t.getPortfolioTransaction().getType() == PortfolioTransaction.Type.BUY) {
+            t.setType(PortfolioTransaction.Type.SELL);
+        }
+        List<Unit> invertedUnits = t.getPortfolioTransaction().getUnits().map((Unit u) -> u.split(-1))
```

Comment:
This is not the indented use of the split method. PP assumes the values are always positive. It is the type of the transactions (or the unit) that tells what to expect. Either TAX or TAX_REFUND - but both values are positives. I know it could be implemented differently, but, hey, that's the way it is now.

Which also unfortunately means the "Storno" is not easy to handle for PP. Usually I would say: just delete the previous purchase order. But this kind of action is not supported by the import. @Nirus2000 do you know how we handled that with other importers?

Reply:
"Stornos" are currently not processed by any importer.

Reply:
Reverted the last commit so this now throws an exception asking the user to delete the original booking.
(Before it would just import a refund order as an actual order…)